server/src/main/java/org/elasticsearch/script/Script.java

```diff
@@ -633,7 +637,8 @@ public XContentBuilder toXContent(XContentBuilder builder, Params builderParams)
         String contentType = options == null ? null : options.get(CONTENT_TYPE_OPTION);
 
         if (type == ScriptType.INLINE) {
-            if (contentType != null && builder.contentType().mediaType().equals(contentType)) {
+            // treating application/json;charset=utf-8 the same as application/json
+            if (contentType != null && (contentType.startsWith(builder.contentType().mediaTypeWithoutParameters()))) {
```

Comment:
related to https://github.com/elastic/elasticsearch/pull/67677/files#r560424309 - although possibly not needed as we will always save `XContentType.JSON.mediaType()`
but can `options.content_type` have a different value than JSON? like YAML?
also - how do I trigger this code? when `Script#toXContent` is used?

@martijnvg I was advised that you might know about this?